### PR TITLE
Fix Regressions caught by CodeQL and Coverity

### PR DESCRIPTION
**Pull request by ryao**
### Motivation and Context
A few recently merged patches introduced regressions.

4c5fec01a48acc184614ab8735e6954961990235 introduced a regression caught by Coverity.

dc5c8006f684b1df3f2d4b6b8c121447d2db0017 introduced a regression caught by CodeQL in a development branch where I am tuning a more aggressive CodeQL configuration than the one we run on pull requests.

### Description
The individual patches have descriptions, but in summary, we switch from `ASSERT3U()` to `ASSERT3S()` in one and add a missing loop counter increment in another.

### How Has This Been Tested?
The buildbot can test it.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by ryao** [comment]
@amotin You might want to re-do your performance evaluation of dc5c8006f684b1df3f2d4b6b8c121447d2db0017 with this fix in place.

@mcmilk Please review the change.

**Review by amotin — APPROVED** [review]

**Comment by ryao** [comment]
I just changed the title of one of the commits. The commit is otherwise identical to the earlier version.

File: module/icp/algs/sha2/sha2_generic.c [hunk 1]
```diff
@@ -400,8 +400,8 @@ SHA2Init(int algotype, SHA2_CTX *ctx)
 	sha256_ctx *ctx256 = &ctx->sha256;
 	sha512_ctx *ctx512 = &ctx->sha512;
 
-	ASSERT3U(algotype, >=, SHA256_MECH_INFO_TYPE);
-	ASSERT3U(algotype, <=, SHA512_256_MECH_INFO_TYPE);
+	ASSERT3S(algotype, >=, SHA256_MECH_INFO_TYPE);
+	ASSERT3S(algotype, <=, SHA512_256_MECH_INFO_TYPE);
```
Comment (mcmilk) [inline module/icp/algs/sha2/sha2_generic.c:405]: @ryao - Oh yes, you are right.

**Review by mcmilk — APPROVED** [review]

**Comment by ryao** [comment]
I changed a commit title and repushed.

**Comment by ryao** [comment]
@behlendorf The zloop failure is a pre-existing issue. I opened #14583 with a fix for it. As for the Fedora issue, I suspect it is a pre-existing issue too, but the telemetry is not good enough for me to see why.

**Comment by ryao** [comment]
The Fedora buildbot failure is https://github.com/openzfs/zfs/issues/14584.

**Pull request closed without merging**